lib/scientist/experiment.rb

```diff
@@ -164,13 +174,9 @@ def name
     "experiment"
   end
 
-  # Internal: compare two observations, using the configured compare block if present.
+  # Internal: compare two observations, using the configured compare and error_compare lambdas if present.
   def observations_are_equivalent?(a, b)
-    if @_scientist_comparator
-      a.equivalent_to?(b, &@_scientist_comparator)
-    else
-      a.equivalent_to? b
-    end
+    a.equivalent_to? b, @_scientist_comparator, @_scientist_error_comparator
```

Comment:
👍 handling nil arguments in the method is easier than this `if` statement in the calling code.